frontend/src/components/auth/register/registerForm/RegisterForm.tsx

```diff
@@ -0,0 +1,169 @@
+
+import React from "react";
+
+import { createStyles, makeStyles } from "@material-ui/core/styles";
+import { Button } from "@material-ui/core";
+
+import useObservable from "../../../../hooks/observable";
+import RegisterError from "./RegisterError";
+import RegisterSuccess from "./RegisterSuccess";
+import EmailField from "./RegisterFormEmail";
+import PasswordField from "./RegisterFormPassword";
+
+import authService from "../../../../services/auth";
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    button: {
+      marginTop: "3%",
+      width: "85%"
+    }
+  })
+);
+
+interface RegisterState {
+  email: string;
+  password: string;
+  password2: string;
+  passwordsComparison: string;
+}
+
+/**
+ * Form with email, password and repeated password
+ */
+function RegisterForm() {
+  const registerInitialState = {
+    email: "",
+    password: "",
+    password2: "",
+    passwordsComparison: ""
+  };
+
+  const [registerData, setRegisterData] = React.useState<RegisterState>(
+    registerInitialState
+  );
+  const [registerErrors, setRegisterErrors] = React.useState<RegisterState>(
+    registerInitialState
+  );
+  const [error, setError] = React.useState();
+  const registeredUser = useObservable(authService.justRegistered());
+
+  /**
+   * Checks if email has correct format
+   *
+   * @param { string } input Email entered by user
+   */
+  const validateEmail = (input: string) => {
+    const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return re.test(String(input).toLowerCase()) ? "" : "Wrong email format.";
+  };
+
+  /**
+   * Checks if password has correct length and if repeated password is identical as password
+   *
+   * @param { string } input Password entered by user
+   */
+  const validatePassword = (input: string) => {
+    if (input.length >= 8) return "";
+    else {
+      console.log("too short");
+      return "Password must be at least 8 characters long.";
+    }
+  };
+
+  /**
+   * Checks if repeated password is identical as password
+   *
+   * @param { string} field password or repeated password input field
+   * @param { string } input Password entered by user
+   */
+  const assertPasswords = (field: string, input: string) => {
+    if (field === "password") {
+      if (input === registerData.password2) return "";
+      else return "Passwords must be identical.";
+    } else if (field === "password2") {
+      if (input === registerData.password) return "";
+      else return "Passwords must be identical.";
+    } else return "";
+  };
+
+  const handleChange = (field: keyof RegisterState) => (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setRegisterData({ ...registerData, [field]: event.target.value });
+    if (field === "email") {
+      setRegisterErrors({
+        ...registerErrors,
+        [field]: validateEmail(event.target.value)
+      });
+    } else {
```

Comment:
Please create object with fields to validators mapping. Now you have the same validation for password and password2 and you can see the same error twice, under two fields. In case of password field you should check only if the length is enough. For password2 you should only check if it matches password